File: ReservationFrontend/src/models/Maintenance.ts
import { MaintenanceType } from "./enums/MaintenanceType";

export interface Maintenance {
  id: number;
  defects: string;
  type: MaintenanceType;
  upcomingServiceNeeds: string;
  startDate: Date;
  plannedEndDate: Date;
  actualEndDate?: Date;
}

File: AnalyticsService/src/main/kotlin/com/rentalcarsystem/analyticsservice/models/Maintenance.kt
package com.rentalcarsystem.analyticsservice.models

import com.rentalcarsystem.analyticsservice.enums.MaintenanceType
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "maintenances")
class Maintenance(
    @Enumerated(EnumType.STRING)
    @Column(nullable = true)
    var type: MaintenanceType,

    @Column(nullable = false)
    var startDate: LocalDateTime,

    @Column(nullable = false)
    var plannedEndDate: LocalDateTime,

    var actualEndDate: LocalDateTime? = null,

    @Column(nullable = false)
    var startFleetManagerUsername: String,

    var endFleetManagerUsername: String? = null

) : BaseEntity<Long>()